packages/use-presence/src/index.tsx

```diff
@@ -107,3 +109,57 @@ export default function usePresence(
     isExiting
   };
 }
+
+export type UseUniqueDataPresenceOptions<T> = {
+  /** Check if two instances of type T are equal, defaults to a === comparison */
+  equalityCheck?(a: T, b: T): boolean;
+  /** Check if an instance of type T is considered valid to render, defaults to Boolean function */
+  validationCheck?(a: T): boolean;
+} & UsePresenceOptions;
+
+function defaultEqualityCheck<T>(a: T, b: T) {
+  return a === b;
+}
+function defaultValidationCheck<T>(a: T) {
+  return Boolean(a);
+}
+
+export function useUniqueDataPresence<T>(
+  dataInput: T,
+  {
+    equalityCheck = defaultEqualityCheck,
+    validationCheck = defaultValidationCheck,
```

Comment:
Here the same question if we can avoid it. Is this intended for the hook to be called with values that are not an item? I think we can just check for `!= null`, if the user needs more customization, also this can be customized by transforming the value that's passed to the hook. Also I think `false` can likely be a valid value.

Reply:
I switched to a non-configurable `isNil` function to validate because I wanted to check `item === null || item === undefined` ℹ️